float32 cell1
float32 cell2
float32 cell3